src/node_contextify.cc

```diff
@@ -1483,50 +1486,109 @@ static MaybeLocal<Function> CompileFunctionForCJSLoader(Environment* env,
   return scope.Escape(fn);
 }
 
+static bool warned_about_require_esm = false;
+// TODO(joyeecheung): this was copied from the warning previously emitted in the
+// JS land, but it's not very helpful. There should be specific information
+// about which file or which package.json to update.
```

Comment:
In JS land, the specific file to update could be determined by looking at the first line of the stack trace, which was right below the warning.

Reply:
That is part of `ERR_REQUIRE_ESM` message, which we plan to get rid of, not part of the warning.

Reply:
Actually for `.js` without a `package.json`, there's not currently a warning on the main branch..but probably out of scope for this PR to fix.